src/solver/clause.rs

```diff
@@ -503,40 +481,33 @@ impl ClauseState {
 /// Represents a literal in a SAT clause (i.e. either A or ¬A)
 #[repr(transparent)]
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
-pub(crate) struct Literal(u32);
+pub(crate) struct Literal(NonZeroU32);
```

Comment:
Dit we mention the autmatic optimization for size that rust does here somewhere?

Reply:
I added a comment, and a test!